dask/dataframe/io/csv.py

```diff
@@ -388,6 +390,31 @@ def text_blocks_to_pandas(
     # Define parts
     parts = []
     colname, paths = path or (None, None)
+
+    if dd._dask_expr_enabled():
+        for i in range(len(blocks)):
+            parts.append([paths[i] if paths else None, is_first[i], is_last[i]])
+
+        return dd.from_map(
+            partial(
+                _read_csv,
+                full_columns=columns,
+                colname=colname,
+                head=head,
+                header=header,
+                reader=reader,
+                dtypes=dtypes,
+                enforce=enforce,
+                kwargs=kwargs,
+                blocksize=blocksize,
+            ),
+            blocks,
+            parts,
+            meta=head,
+            label="read-csv",
```

Comment:
Does this always create task keys with `read-csv`, also for `read_table` and other calls?

Reply:
Yeah :( but it's consistent with the old version for now